packages/css-engine/src/core/rules.ts

```diff
@@ -63,11 +63,25 @@ export type MediaRuleOptions = {
 };
 
 export class MediaRule {
-  #options: MediaRuleOptions;
+  // Sort media rules by minWidth.
+  // Needed to ensure that more specific media rules are inserted after less specific ones.
+  // So that they get a higher specificity.
+  static sort(mediaRules: Iterable<MediaRule>) {
+    return Array.from(mediaRules).sort((a, b) => {
+      if (
+        a.options.minWidth === undefined ||
+        b.options.minWidth === undefined
```

Comment:
So it wouldn't sort but just reverse the below examples, looks like a bug as IMO both arrays after sorting should give same value
```
[{ options: { minWidth: undefined } }, { options: { minWidth: 1 } }]
// and 
[{ options: { minWidth: 1 } }, { options: { minWidth: undefined } }]
```

Reply:
added a test to ensure that